### DISCUSSION TITLE
Meshtastic

### DISCUSSION THREAD

**Discussion by trosel** [markqvist/Reticulum#77] [General]
Someone asked what Reticulum offers that is different from Meshtastic

https://news.ycombinator.com/item?id=32016975

I think there is some huge foundational differences in design, but I’m not sure how to convey that. Anyone want to take a stab at an answer?

**Comment by silveroranges** [comment]
I had never heard of this project before, but from a cursory glance it
seems that their project is focused more on LoRa, and local mesh where as
reticulum is made for any media that can carry a digital signal, and can be
tunneled over the internet or existing infrastructure easily as well as
local mesh networks.

**Comment by markqvist** [comment]
Sorry for being slow to respond here, I was so busy with getting all the latest releases out there.

My knowlegde about Meshtastic is a bit limited, so I might get some details wrong, but from what I understand, Meshtastic is more or less an *application*. It is focused on carrying out a single task well, in a way that is easy to understand and deploy, but that also means that it will not really do anything else than that task, and stop working well if you push it close to, or outside, its intended operational domain.

Reticulum itself is a general purpose communications stack. It will handle any sort of digital communication for you, from sending a single data packet to sending messages or streaming a video call, to serving a library of all human knowledge. But it will not do so itself. Someone will have to *write* the applications that utilise Reticulum to do those tasks. This also means that Reticulum, *in itself*, will not allow anyone to have a chat with someone else. You need an application *using* Reticulum for that, and a Reticulum network.

Right now, only a handful of programs and applications exist that use Reticulum. One of those systems is LXMF, a distributed messaging protocol that enables direct (and offline) messaging between users and machines. There is currently several LXMF clients available that people can use on mobile devices and computers. This is similar to Meshtastic, in that they both serve the same end-goal, of sending messages between users.

I don't know how open or interoperable the Meshtastic protocol is, but I do know that LXMF is very open and easy to integrate in other system, and messages passed with LXMF can be transported over any number of mediums, such as LoRa, WiFi, packet radio or over the Inernet. And you can mix and match the mediums as you please. This is possible since the messaging system is built *on top of* Reticulum (a general purpose networking stack), and it then gains all the capabilities of it (such as efficient multihop routing, global addressing, strong encryption, good privacy and so on).

I hope this clarifies it a bit. If not please ask away and I will try to answer as best I can!

**Reply by markqvist** [reply]
Hmm, I wanted to read the original question on ycombinator and post the above as an asnwer, but for some reason I cannot access their page right now. Maybe the server is just unreachable from my location, or they are having trouble. But if anyone can repost my answer to who asked the question, I'd appreciate it!

**Comment by trosel** [comment]
Here is a review of Meshtastic. It's super helpful to get what that product offers and how the user experienced it 

https://tylercipriani.com/blog/2022/07/31/meshtastic-a-review/
